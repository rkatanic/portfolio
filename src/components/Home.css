.top-shadow {
  position: fixed;
  left: 0;
  height: 10vh;
  z-index: 2;
  width: 100%;
  background: linear-gradient(var(--color-body), transparent);
}

.container {
  position: relative;
  display: flex;
  flex-wrap: wrap;
  max-width: var(--width-desktop);
}

.column-right {
  flex: 1;
  padding: 5vh 2rem;
  position: relative;
}

.column-left {
  top: 0;
  padding: 5vh 2rem;
  position: sticky;
  display: flex;
  justify-content: space-between;
  flex: 1;
  max-height: 100vh;
  flex-direction: column;
  align-items: flex-start;
}

.typewriter-headline {
  font-size: 2.75rem;
  margin: 0;
  color: var(--color-primary);
}

.title {
  margin: 0;
  font-size: 2.5rem;
}

.desc {
  text-align: justify;
  line-height: 2;
}

@media screen and (max-width: 954px) {
  .container {
    flex-direction: column;
  }
  .column-left {
    position: relative;
    max-height: none;
  }
}
