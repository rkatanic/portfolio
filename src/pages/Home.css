.desc {
  text-align: justify;
  font-size: 0.925rem;
  margin: 0;
}

.home-section-title {
  font-size: 1.25rem;
  margin: 0;
}

.home-section-subtitle {
  letter-spacing: 1px;
  font-size: 0.925rem;
  font-weight: 500;
  color: var(--color-text-light);
  margin: 0;
}

.home-section {
  width: 100%;
  padding: 1.5rem;
  margin-bottom: 1rem;
  border-radius: 10px;
  background: var(--color-input);
  box-shadow: 0 1px 3px 0 hsla(0, 0%, 0%, 0.1),
    0 1px 2px 0 hsla(0, 0%, 0%, 0.06);
  position: relative;
  overflow: hidden;
}

.home-section::before {
  position: absolute;
  content: "";
  top: 0;
  left: 0;
  width: 100%;
  height: 6rem;
  background: var(--color-primary);
}

.home-section-row {
  display: flex;
  flex-wrap: wrap;
  gap: 1rem;
  width: 100%;
  flex: 1;
  position: relative;
  margin-bottom: 2rem;
}

.home-section-header {
  text-align: center;
}

.home-section-img {
  width: 8rem;
  height: 8rem;
  background: var(--color-primary);
  border-radius: 10px;
  z-index: 1;
  position: relative;
  border: 4px solid var(--color-input);
}

.home-section-info {
  margin-bottom: 1rem;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: end;
}

.home-section-info-item {
  display: flex;
  justify-content: space-between;
  margin: 0.75rem 0 0;
}

.home-section-info-item span:first-child {
  color: var(--color-text-light);
}
