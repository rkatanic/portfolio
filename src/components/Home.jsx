import Section from "./Section";
import myimage from "../assets/images/me1.png";
import { ReactComponent as MailSendIcon } from "../assets/icons/mail-send.svg";

import "./Home.css";

const Home = () => {
  return (
    <Section id="home">
      <div className="home">
        <div>
          <h2 className="home-title">Hi, I'm Radivoje</h2>
          <h3 className="home-subtitle">Full Stack Developer</h3>
          <p className="home-description">
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quaerat
            aperiam nam unde sunt dicta error doloribus est ex animi reiciendis
            necessitatibus debitis nulla, nemo quis earum aliquid perferendis
            esse iure!
          </p>
          <button type="button" className="btn-primary">
            Contact
            <MailSendIcon />
          </button>
        </div>
        <div className="home-image-container">
          <img className="home-img" src={myimage} alt="my-img" />
        </div>
      </div>
    </Section>
  );
};

export default Home;
