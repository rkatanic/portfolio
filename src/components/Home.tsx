import Section from "./Section";
import Button from "./Button";
import Typewriter, { TypewriterClass } from "typewriter-effect";
import myimage from "../assets/images/me1.png";
import { ReactComponent as MailSendIcon } from "../assets/icons/mail-send.svg";
import { ReactComponent as ArrowIcon } from "../assets/icons/arrow-up.svg";

import "./Home.css";
import { useEffect, useState } from "react";

const Home = (): JSX.Element => {
  let a = 0;
  let b = 0;
  let c = 0;
  let d = 0;
  let e = 0;
  let f = 0;
  let g = 0;
  let h = 0;
  let i = 0;
  let j = 0;
  const textOne = "fetching data for Radivoje ...";
  const textTwo = "Radivoje successfully found!";
  const textThree = "// Full name: Radivoje Katanic";
  const textFour = "// Age: 25";
  const textFive = "// Location: Doboj";
  const textSix = "processing some more information ...";
  const textSeven = "Additional information";
  const textEight = "// Role: Software Developer";
  const textNine =
    "// Description: Lorem, ipsum dolor sit amet consectetur adipisicing elit. Similique ducimus doloremque unde dicta consectetur fugiat, magni ipsa nihil esse vel eaque reprehenderit odio, maxime soluta tempora repellat cum possimus facere.";

  const speedOne = 50;
  const speedTwo = 5;
  const speedThree = 1;

  const typeWriterOne = () => {
    if (a < textOne.length) {
      document.getElementById("typer-one")!.innerHTML += textOne.charAt(a);
      a++;
      setTimeout(typeWriterOne, speedOne);
    }
  };

  const typeWriterTwo = () => {
    if (b < textTwo.length) {
      document.getElementById("typer-two")!.innerHTML += textTwo.charAt(b);
      b++;
      setTimeout(typeWriterTwo, speedTwo);
    }
  };

  const typeWriterThree = () => {
    if (c < textThree.length) {
      document.getElementById("typer-three")!.innerHTML += textThree.charAt(c);
      c++;
      setTimeout(typeWriterThree, speedOne);
    }
  };

  const typeWriterFour = () => {
    if (d < textFour.length) {
      document.getElementById("typer-four")!.innerHTML += textFour.charAt(d);
      d++;
      setTimeout(typeWriterFour, speedOne);
    }
  };

  const typeWriterFive = () => {
    if (e < textFive.length) {
      document.getElementById("typer-five")!.innerHTML += textFive.charAt(e);
      e++;
      setTimeout(typeWriterFive, speedTwo);
    }
  };

  const typeWriterSix = () => {
    if (f == 0) {
      let prefix = document.createElement("b");
      prefix.className = "color-primary";
      prefix.innerHTML = "$> ";
      document.getElementById("typer-six")?.appendChild(prefix);
    }
    if (f < textSix.length) {
      document.getElementById("typer-six")!.innerHTML += textSix.charAt(f);
      f++;
      setTimeout(typeWriterSix, speedTwo);
    }
  };

  const typeWriterSeven = () => {
    if (g < textSeven.length) {
      document.getElementById("typer-seven")!.innerHTML += textSeven.charAt(g);
      g++;
      setTimeout(typeWriterSeven, speedThree);
    }
  };

  const typeWriterEight = () => {
    if (h < textEight.length) {
      document.getElementById("typer-eight")!.innerHTML += textEight.charAt(h);
      h++;
      setTimeout(typeWriterEight, speedOne);
    }
  };

  const typeWriterNine = () => {
    if (i < textNine.length) {
      document.getElementById("typer-nine")!.innerHTML += textNine.charAt(i);
      i++;
      setTimeout(typeWriterNine, speedTwo);
    }
  };
  useEffect(() => {
    setTimeout(() => typeWriterOne(), 500);
    setTimeout(() => typeWriterTwo(), 3500);
    setTimeout(() => typeWriterThree(), 4000);
    setTimeout(() => typeWriterFour(), 5500);
    setTimeout(() => typeWriterFive(), 6500);
    setTimeout(() => typeWriterSix(), 7000);
    setTimeout(() => typeWriterSeven(), 7300);
    setTimeout(() => typeWriterEight(), 8000);
    setTimeout(() => typeWriterNine(), 8200);
  }, []);

  return (
    <Section id="home">
      <div className="home">
        <p className="comment white" id="typer-one">
          <b className="color-primary">${"> "}</b>
        </p>
        <h3 className="color-primary" id="typer-two"></h3>

        <p className="comment" id="typer-three"></p>
        <p className="comment" id="typer-four"></p>
        <p className="comment" id="typer-five"></p>
        <p className="comment white" id="typer-six"></p>

        <h3 className="color-primary" id="typer-seven"></h3>
        <p className="comment" id="typer-eight"></p>
        <p className="comment" id="typer-nine"></p>

        {/*  <div className="home-content">
          <h3 className="home-title">
            I am&nbsp;
            <Typewriter
              options={{ loop: true }}
              onInit={(typewriter) => {
                typewriter
                  .start()
                  .typeString("a software developer.")
                  .deleteAll()
                  .typeString("a coder")
                  .deleteAll();
              }}
            />
          </h3>
          <h3 className="home-subtitle">Software Developer</h3>
          <p className="home-description">
            My main focus is on building efficient web apps that work across all
            platforms and browsers. Building web solutions is my satisfaction.
            Sometimes a problem solving can be relaxing too.
          </p>
          <Button
            onClick={() => document.getElementById("contact")?.scrollIntoView()}
          >
            <span> Contact me</span>
            <MailSendIcon />
          </Button>
          <a className="scroll-down" href="#about">
            <div className="scroll-down-container">
              <div className="scroll-down-bubble"></div>
            </div>
            Read more <ArrowIcon />
          </a>
        </div> */}
      </div>
    </Section>
  );
};

export default Home;
