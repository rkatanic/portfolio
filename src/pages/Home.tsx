import Typewriter from "typewriter-effect";
import Navigation from "../components/Navigation";
import ProgressBar from "../components/ProgressBar";
import Footer from "../components/Footer";

import "./Home.css";

const Home = (): JSX.Element => (
  <>
    <div>
      <ProgressBar />
      <h3 className="title">
        Hello, I<span className="primary">'</span>m
        <br /> Radivoje{" "}
        <span className="last-name">
          Katanic.<span className="apostrophe primary">'</span>{" "}
        </span>
      </h3>
      <h3 className="typewriter-headline">
        <Typewriter
          options={{
            cursor: "_",
            strings: ["Web Developer", "Coder"],
            autoStart: true,
            loop: true,
          }}
        />
      </h3>
      <p className="desc">
        I am a full stack developer located in Doboj, Bosnia & Herzegovina.
        Currently I'm crafting software products at{" "}
        <a
          className="home-link"
          href="https://productdock.com/"
          target="_blank"
          rel="noopener noreferrer"
        >
          ProductDock
        </a>
        .
        <br />I do what coders do. I like what coders like. In the end of the
        day, I'm just a regular person who likes doing some sports in free time,
        watching movies and reading all kinds of articles.
      </p>
    </div>
    <Navigation />
    <Footer />
  </>
);

export default Home;
