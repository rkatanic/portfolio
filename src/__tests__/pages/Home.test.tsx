import { render } from "@testing-library/react";
import Home from "../../pages/Home";

describe("Home", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Home />);

    expect(baseElement).toMatchSnapshot();
  });
});
