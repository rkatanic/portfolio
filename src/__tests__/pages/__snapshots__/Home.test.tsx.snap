// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<body>
  <div>
    <div>
      <div
        class="progress-bar"
      >
        <div
          class="progress-bar-pulse"
        />
        <div
          class="progress-bar-percent"
          style="height: 0%;"
        />
      </div>
      <h3
        class="title"
      >
        Hello, I
        <span
          class="primary"
        >
          '
        </span>
        m
        <br />
         Radivoje
         
        <span
          class="last-name"
        >
          Katanic.
          <span
            class="apostrophe primary"
          >
            '
          </span>
           
        </span>
      </h3>
      <h3
        class="typewriter-headline"
      >
        <div
          class="Typewriter"
          data-testid="typewriter-wrapper"
        >
          <span
            class="Typewriter__wrapper"
          />
          <span
            class="Typewriter__cursor"
          >
            _
          </span>
        </div>
      </h3>
      <p
        class="desc"
      >
        I am a full stack developer located in Doboj, Bosnia & Herzegovina. Currently I'm crafting software products at
         
        <a
          class="home-link"
          href="https://productdock.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          ProductDock
        </a>
        .
        <br />
        I do what coders do. I like what coders like. In the end of the day, I'm just a regular person who likes doing some sports in free time, watching movies and reading all kinds of articles.
      </p>
    </div>
    <nav
      class="nav"
    >
      <div
        class="nav-menu"
        data-testid="nav-menu"
      >
        <ul
          class="nav-list"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#about"
            >
              <div
                class="number"
              >
                01
              </div>
              <div
                class="dash"
              />
              <span>
                About
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#skills"
            >
              <div
                class="number"
              >
                02
              </div>
              <div
                class="dash"
              />
              <span>
                Skills
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#faq"
            >
              <div
                class="number"
              >
                03
              </div>
              <div
                class="dash"
              />
              <span>
                FAQ
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#contact"
            >
              <div
                class="number"
              >
                04
              </div>
              <div
                class="dash"
              />
              <span>
                Contact
              </span>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="nav-toggle"
        data-testid="nav-toggle"
      >
        <div
          class="nav-toggle-dash"
        />
      </div>
    </nav>
    <div
      class="footer"
    >
      <div
        class="footer-content"
      >
        <img
          alt=""
          class="footer-image"
          src="me1.png"
        />
        <div
          class="footer-links"
        >
          <a
            class="footer-link"
            href="https://www.linkedin.com/in/radivoje-katanic"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="footer-link-icon"
            >
              <svg>
                linkedin.svg
              </svg>
            </div>
            LinkedIn
          </a>
          <a
            class="footer-link"
            href="https://github.com/k-rade"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="footer-link-icon"
            >
              <svg>
                github.svg
              </svg>
            </div>
            GitHub
          </a>
        </div>
      </div>
      <div
        class="footer-actions"
      >
        <div
          class="color-switch"
        >
          <div
            class="color-switch-header"
          >
            <svg>
              pallete.svg
            </svg>
          </div>
        </div>
        <div
          class="night-mode-switch"
        >
          <svg>
            sun.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
